app.rb

```diff
@@ -0,0 +1,104 @@
+# rubocop:disable Metrics/MethodLength
+
+require_relative './teacher'
+require_relative './student'
+require_relative './person'
+require_relative './book'
+
+class App
+  def initialize
+    @rentals = []
+    @people = []
+    @books = []
+  end
+
+  def all_books
+    if @books.length.positive?
+      @books.each { |book| puts "Title: \"#{book.title}\" Author: #{book.author}" }
+    else
+      puts 'No books in library'
+    end
+  end
+
+  def all_people
+    if @people.length.positive?
+      @people.each { |person| puts "Name: \"#{person.name}\" Age: #{person.age}" }
+    else
+      puts 'No people in library'
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)?'
+    print 'Select (1) or (2): '
+    selection = gets.chomp
+
+    case selection
+    when '1'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Has parent permission? [Y/N]: '
+      permission = gets.chomp.downcase
+      permission = permission == 'y'
+      @people << Student.new(age, name, permission)
+      puts 'student created successfully'
+    when '2'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Specialization: '
+      specialization = gets.chomp
+      @people << Teacher.new(age, specialization, name)
+      puts 'Teacher created successfully'
+    else
+      puts 'invalid selection'
+      nil
+    end
+  end
```

Comment:
While this is a very smart method you have over here 🤓, still needs to be refactored. 🤔

Why❓ 
All requests for input and input validation should've been placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.

Your method is dealing with two different responsibilities... Breaking the first rule in SOLID principles  *Single-responsibility Principle*

How❓
Try to break it into two different single responsibility methods and move it to your `main.rb` file, and for your case it will be in `action` method in `helpers.rb` file